tpm2/test/src/cipher/aes_gcm_test.c

```diff
@@ -56,6 +63,411 @@ int aes_gcm_add_tests(CU_pSuite suite)
+        // after either completing the six-step parsing procedure or
+        // encountering an unexpected input line, return to initial step
+        step = 1;
+      }
+      else if ((strncmp(buffer, "FAIL", 4) == 0) &&
+               (step == 6))
+      {
+        // process 'FAIL' result component of vector
+        PT_len = 0;
+        pass_result = false;
+
+        // check applicability of parsed vector to kmyth implementation
+        //   - kmyth does not support additional authenticated data (AAD)
+        //   - kmyth uses hard-coded IV length (GCM_IV_LEN)
+        //   - kmyth uses hard-coded Tag length (GCM_TAG_LEN)
+        if ((AAD_len == 0) &&
+            (IV_len == GCM_IV_LEN) &&
+            (Tag_len == GCM_TAG_LEN))
+        {
+            test_vector_found = true;
+        }
+        // after either completing the six-step parsing procedure or
+        // encountering an unexpected input line, return to initial step
+        step = 1;
+      }
+      else
+      {
+        step = 1;
+      }
+    }
+  }
+
+  if (test_vector_found)
+  {
+    // copy parsed and validated test vector information to output parameters
+    memcpy(*key_vec, Key, Key_len);
+    *key_vec_len = Key_len;
+    memcpy(*input_vec, IV, IV_len);
+    memcpy(*input_vec + IV_len, CT, CT_len);
+    memcpy(*input_vec + IV_len + CT_len, Tag, Tag_len);
+    *input_vec_len = IV_len + CT_len + Tag_len;
+    memcpy(*result_vec, PT, PT_len);
+    *result_vec_len = PT_len;
+    *expect_pass = pass_result;
+  }
+
+  // clean-up memory allocated to buffer test vector(s)
+  free(Key);
+  free(IV);
+  free(AAD);
+  free(Tag);
+  free(CT);
+  free(PT);
+
+  // if while loop exit due to EOF, return unsuccessful result
+  if (!test_vector_found)
+  {
+    return 1;
+  }
+
+  return 0;
+}
+
+
+//----------------------------------------------------------------------------
+// test_aes_gcm_decrypt_vectors()
+//----------------------------------------------------------------------------
+void test_aes_gcm_decrypt_vectors(void)
+{
+  // specify the compilation of test vector mappings for kmyth AES GCM
+  // decrypt cipher testing.
+  const cipher_vector_compilation gcm_decrypt_vectors = {
+    .count = 3,
+    .sets = 
+    {
+      { .desc = "AES-128, Galois Counter Mode (GCM), decryption",
+        .func_to_test = "aes_gcm_decrypt",
+        .path = "./test/data/gcmtestvectors/gcmDecrypt128.rsp" },
+      { .desc = "AES-192, Galois Counter Mode (GCM), decryption",
+        .func_to_test = "aes_gcm_decrypt",
+        .path = "./test/data/gcmtestvectors/gcmDecrypt192.rsp" },
+      { .desc = "AES-256, Galois Counter Mode (GCM), decryption",
+        .func_to_test = "aes_gcm_decrypt",
+        .path = "./test/data/gcmtestvectors/gcmDecrypt256.rsp" }
+    }
+  };
+
+  // array of file pointers for test vector files
+  FILE * test_vector_fd[MAX_VECTOR_SETS_IN_COMPILATION] = {NULL};
+
+  // counter to track number of test vectors applied from a file
+  int test_vector_count = 0;
```

Comment:
I think it's cleaner if this and `done_with_test_vector_file` are local to the loop over the files.

Reply:
I moved the declaration/initialization for test_vector_count and done_with_test_vector_file right before the while loop iteratively processing the file.